File: src/java/nz/co/searchwellington/controllers/ShowBrokenDecisionService.java
package nz.co.searchwellington.controllers;

import nz.co.searchwellington.model.User;

public class ShowBrokenDecisionService {

	private LoggedInUserFilter loggedInUserFilter;
	
	
	public ShowBrokenDecisionService(LoggedInUserFilter loggedInUserFilter) {
		this.loggedInUserFilter = loggedInUserFilter;
	}


	public boolean shouldShowBroken() {
		User loggedInUser = loggedInUserFilter.getLoggedInUser();
		return loggedInUser != null && loggedInUser.isAdmin();
	}

}

File: src/main/scala/nz/co/searchwellington/controllers/ShowBrokenDecisionService.scala
package nz.co.searchwellington.controllers

import nz.co.searchwellington.model.User
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component

@Component class ShowBrokenDecisionService @Autowired()() {

  def shouldShowBroken(loggedInUser: Option[User]): Boolean = {
    loggedInUser.exists { user =>
      user.isAdmin
    }
  }
}
